WebApp/src/main/java/ev/projects/webapp/schedulers/CaseStatsScheduler.java

```diff
@@ -0,0 +1,75 @@
+package ev.projects.webapp.schedulers;
+
+import ev.projects.models.Case;
+import ev.projects.models.Document;
+import ev.projects.models.Report;
+import ev.projects.services.ICaseService;
+import ev.projects.services.IDocumentService;
+import ev.projects.services.IReportService;
+import lombok.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.List;
+
+@Component
+public class CaseStatsScheduler {
+
+    @Autowired
+    private ICaseService caseService;
+    @Autowired
+    private IReportService reportService;
+    @Autowired
+    private IDocumentService documentService;
+
+    @Scheduled(fixedRateString = "${scheduler.rate}")
+    public void logStats() {
+        CaseStats caseStats = findBiggestCase();
+        Case biggestCase = caseStats.getBiggestCase();
+        if(biggestCase != null) {
+            String dataToLog = "Biggest case has size of (" + caseStats.getSize() +
+                    " bytes) and following data:" + biggestCase.toString();
+            Report report = new Report(0, new Date(), dataToLog);
+            reportService.add(report);
+        }
+    }
+
+    public CaseStats findBiggestCase() {
```

Comment:
method should be `private` unless used outside the class

Reply:
Missed it, thanks. Fixed in 9dc2176.